Formula/m/mdz.rb

```diff
@@ -0,0 +1,26 @@
+class Mdz < Formula
+  desc "CLI for the mdz ledger Open Source"
+  homepage "https://github.com/LerianStudio/midaz"
+  url "https://github.com/LerianStudio/midaz/archive/refs/tags/v1.30.0.tar.gz"
+  sha256 "be3d5befc9be00119257c63e47051042e594770706e1e3c16e453dc2bc806ca2"
+  license "Apache-2.0 license"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/LerianStudio/midaz/components/mdz/pkg/environment.ClientID=9670e0ca55a29a466d31
+      -X github.com/LerianStudio/midaz/components/mdz/pkg/environment.ClientSecret=dd03f916cacf4a98c6a413d9c38ba102dce436a9
+      -X github.com/LerianStudio/midaz/components/mdz/pkg/environment.URLAPIAuth=http://127.0.0.1:8080
+      -X github.com/LerianStudio/midaz/components/mdz/pkg/environment.URLAPILedger=http://127.0.0.1:3000
+      -X github.com/LerianStudio/midaz/components/mdz/pkg/environment.Version=#{version}
+    ]
+
+    system "go", "build", *std_go_args(ldflags:), "./components/mdz/main.go"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/azion --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate tenew-formula-rapidyamlath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks, I'll be updating!